radix-engine/src/kernel/call_frame.rs

```diff
@@ -686,21 +702,26 @@ impl<L: Clone> CallFrame<L> {
         }
 
         if let Some(store_handle) = store_handle {
-            Ok(store.update_substate(*store_handle, substate))
+            store.update_substate(*store_handle, substate);
         } else {
             heap.set_substate(*node_id, *partition_num, substate_key.clone(), substate);
-            Ok(StoreAccessInfo::new())
         }
+
+        Ok(())
     }
 
-    pub fn create_node<'f, S: SubstateStore>(
+    pub fn create_node<'f, S: SubstateStore, E>(
         &mut self,
         node_id: NodeId,
         node_substates: NodeSubstates,
         heap: &mut Heap,
         store: &'f mut S,
-        handler: &mut impl CallFrameEventHandler,
-    ) -> Result<StoreAccessInfo, CreateNodeError> {
+        handler: &mut impl CallFrameEventHandler<L, E>,
+    ) -> Result<(), CallbackError<CreateNodeError, E>> {
+        // TODO: We need to protect transient blueprints from being globalized directly
+        // into store. This isn't a problem for now since only native objects are allowed
+        // to be transient.
```

Comment:
Nice spot!

Easier to fix it than leaving a task here?
```
if push_to_store {
    handler.on_persist_node(heap, node_id).map_err(CallbackError::CallbackError)?;
}
```

Reply:
Definitely another PR, I think this may be a little more involved than just that line since I don't think the node exists in the heap space which would be required.